### PR TITLE
BI-2 986: Mark exists=False for trashed items 

### PR DESCRIPTION
**Pull request by reed-atakama**
Mark exists as False only if they are removed. This PR also removes the additional api hit in the `events` function. This will leave `path` as `None` for the event and should wait to get populated by `get_latest` when syncing.

This is blocked by the corresponding cloudsync changes: https://github.com/AtakamaLLC/cloudsync/pull/329

### CODE REVIEW

File: cloudsync_gdrive.py [hunk 1]
```diff
@@ -337,6 +337,9 @@ def events(self) -> Generator[Event, None, None]:  # pylint: disable=too-many-lo
 
                 ohash = None
                 path = self._path_oid(oid, use_cache=False)
```
Comment (reed-atakama) [inline cloudsync_gdrive.py:339]: Another option would be calling `info_oid` here, grabbing the path from info and adding the `trashed` flag to info

**Review by earthman1 — DISMISSED** [review]

File: cloudsync_gdrive.py (hunk 1 above)
Comment (earonesty) [inline cloudsync_gdrive.py:339]: wow this is super expensive... we double-tap gdrive on every event.  need to remove that stuff.   should just not include the path.

File: tests/test_provider.py [hunk 2]
```diff
@@ -7,6 +9,70 @@
...
+                'file': { 'mimeType': provider._folder_mime_type }, 'removed': False}]}
+        else:
+            return api(resource, method, *args, **kwargs)
+
+    with patch.object(provider, "_api", side_effect=patched_api):
+        events = list(provider.events())
+
+    assert len(events) == 1
+    # Mark trashed folder non existant
+    assert not events[0].exists
```
[the remark this answers is not in the document]
Comment (reed-atakama) [reply tests/test_provider.py:74]: You you're right, I didn't update this when the change was made to mark exists as None

**Review by earthman1 — APPROVED** [review]
Please make sure there is a test for this behavior in cloudsync also

**Pull request merged**